tools/rofiTool/preview.cpp

```diff
@@ -1,27 +1,40 @@
-#include "commands.hpp"
-#include "rendering.hpp"
-
 #include <fstream>
 #include <stdexcept>
 
 #include <configuration/rofiworld.hpp>
+#include <dimcli/cli.h>
+
+#include "common.hpp"
+#include "rendering.hpp"
 
-static auto command = Dim::Cli().command( "preview" )
-    .desc( "Interactively preview a configuration" );
-static auto& inputFile = command.opt< std::string >( "<FILE>" )
-    .desc("Specify source file");
 
-int preview( Dim::Cli & /* cli */ ) {
-    auto configuration = parseConfiguration( *inputFile );
+void preview( Dim::Cli & cli );
 
-    if ( configuration.modules().size() == 0 )
-        throw std::runtime_error( "Configuration in '" + *inputFile + "' does not contain any modules to display" );
+static auto command = Dim::Cli()
+                              .command( "preview" )
+                              .action( preview )
+                              .desc( "Interactively preview a rofi world" );
+static auto & inputFile = command.opt< std::string >( "<world_file>" )
+                                  .desc( "Specify rofi world source file" );
 
-    affixConfiguration( configuration );
+void preview( Dim::Cli & cli )
+{
+    auto world = parseRofiWorld( *inputFile );
+    if ( !world ) {
+        cli.fail( EXIT_FAILURE, "Error while parsing world", world.assume_error() );
+        return;
+    }
+    if ( world->modules().empty() ) {
+        cli.fail( EXIT_FAILURE, "Empty rofi world" );
+        return;
+    }
 
-    configuration.prepare().get_or_throw_as< std::runtime_error >();
+    affixRofiWorld( *world );
```

Comment:
Affixing a world in space (meaning adding a RigidJoint between some module and 0,0,0) should be done only if the world does not already contain a SpaceJoint ( `if ( world->referencePoints().empty() )` ), otherwise `.prepare()` and/or `.validate()` should not need it.

Reply:
I was aware of the problem, but I didn't think yet of a solution - and this is a nice one. (I was thinking about something like a flag...)